src/main_controller.ts

```diff
@@ -301,31 +301,51 @@ export class MainController implements vscode.Disposable {
     };
 
     private async checkNeovimVersion(): Promise<void> {
-        const minVersion = "0.9.2";
-
-        const [minMajor, minMinor, minPatch] = minVersion.split(".").map((n) => +n);
-        const [, info] = await this.client.apiInfo;
-        const { major, minor, patch } = info.version;
-
-        let outdated = false;
-        if (!info.ui_events.find((e) => e.name === "win_viewport")) {
-            outdated = true;
-        } else if (major < minMajor) {
-            outdated = true;
-        } else if (major === minMajor) {
-            if (minor < minMinor) {
-                outdated = true;
-            } else if (minor === minMinor) {
-                if (patch < minPatch) {
-                    outdated = true;
-                }
+        const minVersion = [0, 9, 0];
+        // It is necessary to check the functionalities added in the minimum version
+        // because users might be using the development version with incomplete functionalities.
+        const requirements = {
+            options: ["statuscolumn"],
+            functions: ["nvim_exec2"],
+        };
+        // check version
+        {
+            const [, info] = await this.client.apiInfo;
+            const { major, minor, patch } = info.version;
+            const currVersion = [major, minor, patch];
+            if (minVersion.some((v, i) => v > currVersion[i])) {
```

Comment:
This will reject version 0.10.1 if minVersion is 0.9.2

Reply:
My bad